ddtrace/internal/core.py

```diff
@@ -162,20 +182,16 @@ def dispatch(self, event_id, args, *other_args):
                     "must be passed in a list. For example, use dispatch('foo', [[l1, l2], arg2]) "
                     "instead of dispatch('foo', [l1, l2], arg2)."
                 )
-        results = []
-        exceptions = []
-        for listener in self._listeners.get(event_id, []):
-            result = None
-            exception = None
+        results = EventResultDict()
+        for name, listener in reversed(list(self._listeners[event_id].items())):
```

Comment:
The problem is already present in the current state, so it only works right now because the registration order of some listener is ok on the same event. I can split the problematic event, this is indeed the best move.